modules/test-runner/src/util/channelProvider.ts

```diff
@@ -0,0 +1,197 @@
+import {
+  ChannelProviderConfig,
+  ChannelProviderRpcMethod,
+  IChannelProvider,
+  IConnextClient,
+  IRpcConnection,
+  JsonRpcRequest,
+  StorePair,
+} from "@connext/types";
+import EventEmitter from "events";
+
+export const createChannelProvider = async (channel: IConnextClient): Promise<IChannelProvider> => {
+  const connection = new RpcConnection(channel);
+  const channelProvider = new ChannelProvider(connection);
+  await channelProvider.enable();
+  return channelProvider;
+};
+
+export class RpcConnection extends EventEmitter implements IRpcConnection {
+  public connected: boolean = true;
+  public channel: IConnextClient;
+
+  constructor(channel: IConnextClient) {
+    super();
+    this.channel = channel;
+  }
+
+  public async send(payload: any): Promise<any> {
+    if (!this.connected) {
+      // IRL this would take 30s to throw
+      throw new Error("RpcConnection: Timeout - JSON-RPC not responded within 30s");
+    }
+    const result = await this.channel.channelProvider.send(payload.method, payload.params);
+    return result;
+  }
+
+  public open(): void {
+    this.connected = true;
+  }
+
+  public close(): void {
+    this.connected = false;
+  }
+}
+
+export class ChannelProvider extends EventEmitter implements IChannelProvider {
```

Comment:
can we name this `MockChannelProvider` or something if it's only going to be used for tests?